logstash-core/spec/logstash/pipelines_registry_spec.rb

```diff
@@ -101,6 +101,39 @@
         end
       end
     end
+
+    context "when pipeline is initializing" do
+      let (:wait_start_create_block) { Queue.new }
+      let (:wait_before_exiting_create_block) { Queue.new }
+      let (:slow_initializing_pipeline) { double("slow_initializing_pipeline") }
+      let (:pipeline2) { double("pipeline2") }
+
+      it "should create a loading state before calling the create block" do
```

Comment:
The test description can be confusing since it mentions that there should be a loading state, but then there's no other reference to asserting that the state is correct.

Reply:
added more comments. the only way to test without adding more API just for testing is to use indirect testing.